### PR TITLE
Bump backoff/v3 to backoff/v4

### PR DESCRIPTION
**Pull request by crazybolillo**
This makes it so projects that are already using backoff don't need to pull up an old version of it just because of vault/api. This major version change is compatible with existing usage in code.

Refs #24707.

### CODE REVIEW

**Comment by VioletHynes** [comment]
Hey @crazybolillo ! I'm interested in getting this merged. I just tested this a little locally and it doesn't look like it breaks anything.

Before we can do that, it looks like there's some merge conflicts. Would it be possible for you to resolve them? Once those get resolved, I'd be happy to approve and merge this in.

**Comment by crazybolillo** [comment]
Thanks, I have rebased the MR and fixed the conflicts.

**Review by VioletHynes — APPROVED** [review]
This looks good to me! I'll wait for CI to run and for everything to be green, and I'll try to get it merged in after.

**Pull request merged**
